projects/steering/src/steering_digital_input.c

```diff
@@ -0,0 +1,65 @@
+#include "steering_digital_input.h"
+#include "event_queue.h"
+#include "gpio_it.h"
+#include "interrupt_def.h"
+#include "soft_timer.h"
+#include "status.h"
+#include "steering_can.h"
+#include "steering_events.h"
+
+GpioAddress s_steering_address_lookup_table[NUM_STEERING_DIGITAL_INPUTS] = {
+  [STEERING_DIGITAL_INPUT_HORN] = { .port = GPIO_PORT_B, .pin = 1 },
+  [STEERING_DIGITAL_INPUT_RADIO_PPT] = { .port = GPIO_PORT_A, .pin = 6 },
+  [STEERING_DIGITAL_INPUT_HIGH_BEAM_FORWARD] = { .port = GPIO_PORT_A, .pin = 7 },
+  [STEERING_DIGITAL_INPUT_HIGH_BEAM_REAR] = { .port = GPIO_PORT_B, .pin = 0 },
+  [STEERING_DIGITAL_INPUT_REGEN_BRAKE_TOGGLE] = { .port = GPIO_PORT_A, .pin = 4 },
+  [STEERING_DIGITAL_INPUT_CC_TOGGLE] = { .port = GPIO_PORT_A, .pin = 5 },
+};
+
+EventId s_steering_event_lookup_table[NUM_STEERING_DIGITAL_INPUTS] = {
+  [STEERING_DIGITAL_INPUT_HORN] = STEERING_INPUT_HORN_EVENT,
+  [STEERING_DIGITAL_INPUT_RADIO_PPT] = STEERING_RADIO_PPT_EVENT,
+  [STEERING_DIGITAL_INPUT_HIGH_BEAM_FORWARD] = STEERING_HIGH_BEAM_FORWARD_EVENT,
+  [STEERING_DIGITAL_INPUT_HIGH_BEAM_REAR] = STEERING_HIGH_BEAM_REAR_EVENT,
+  [STEERING_DIGITAL_INPUT_REGEN_BRAKE_TOGGLE] = STEERING_REGEN_BRAKE_EVENT,
+  [STEERING_DIGITAL_INPUT_CC_TOGGLE] = STEERING_INPUT_CC_TOGGLE_PRESSED_EVENT
+};
+
+GpioAddress *test_get_address(int digital_input_id) {

```

Comment:
If I did this the implementation would be messed up since I would need to manually set up interrupts for each one. Also, if I remember correctly Arshan told me to put the lookup table

Reply:
Couldn't you still have the lookup table, just also have the #defines? this way you can keep your implementation but not have the helper function

Reply:
Also I tried this and its not working

Reply:
hmm it's used this way in power distro inc/power_distribution_pin_defs.h: 
```
#define POWER_DISTRIBUTION_I2C_SCL_PIN \
  { GPIO_PORT_B, 10 }
```
maybe try that?